### DISCUSSION TITLE
What's the best way to track implementation of tracing in Vector?

### DISCUSSION THREAD

**Discussion by kamalmarhubi** [vectordotdev/vector#13490] [General]
I'm subscribed to several issues that seem like they should be related, but I feel like I might be missing an umbrella issue or milestone or project.

Ones I'm subscribed to:
- #2661
- #3949
- #4063
- #1444

**Comment by jszwedko** [comment]
Hi @kamalmarhubi !

Good question. This actually brought my attention to a few stale issues, as well, so I closed a few of the ones you have linked there 🙂

We recently added support for ingesting traces from the Datadog Agent in the `datadog_agent` source and sending them to Datadog via the `datadog_traces` sink. This laid the foundations for handling traces in Vector including an internal data model for it (that matches the OpenTelemetry traces data model).

Past this, the additional work is just adding additional integrations and more processing support, for example an `opentelemetry` source and sink (tracked by https://github.com/vectordotdev/vector/issues/1444) or converting logs to spans (which you opened 😄 https://github.com/vectordotdev/vector/issues/13260) .

As such there isn't an umbrella issue because there is no definition of "finished" for tracing support, just extending functionality. Is there any particular functionality you are interested in, though?

**Reply by ELLIOTTCABLE** [reply]
We're also interested in this — we started integrating vector as a part of our new 'integrate observability via OpenTelemetry' work, but feels like a pretty hard wall to hit to discover that Vector doesn't even support traces … aren't these ~50% of the purpose of OTel? 😩

(The specific tooling that we're investigating at this particular moment looks something like: Vector, Clickhouse as storage, and some combination of SigNoz/Grafana/Honeycomb for visualization and analysis. We also have ES/Kibana in the mix, and a lot of our logging and analytics hang out there, although I'd like to move away from that …)

I saw [this RFC](https://github.com/vectordotdev/vector/blob/master/rfcs/2022-03-15-11851-ingest-opentelemetry-traces.md) from early 2022, but it doesn't seem like there's been any progress on it, unless I missed a PR somewhere?

**Reply by jszwedko** [reply]
Hi @ELLIOTTCABLE !

This is an an area we would like to invest more in. Unfortunately while traces are probably the most robust part of the OpenTelemetry ecosystem they are the weakest part of Vector (which best supports Logs followed by Metrics and then Traces). We would like to get to it, but don't currently have a timeline. We would, of course, support contributions in this area.
